test/parallel/test-net-persistent-nodelay.js

```diff
@@ -0,0 +1,43 @@
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+var serverConnection;
+var echoServer = net.createServer(function(connection) {
+  serverConnection = connection;
+  connection.end()
+});
+echoServer.listen(common.PORT);
+
+var setTrue = false;
+
+var Socket = net.Socket;
+
+Socket.prototype.setNoDelay = function(enable) {
+  this._setNoDelay = enable === undefined ? true : !!enable;
+  if (this._handle && this._handle.setNoDelay) {
+    if (enable === undefined || !!enable) {
+      setTrue = true;
+    }
+    this._handle.setNoDelay(this._setNoDelay);
+  }
+}
+
+echoServer.on('listening', function() {
+  var sock1 = new Socket({});
+  // setNoDelay before the handle is created
+  // there is probably a better way to test this
+
+  sock1.setNoDelay();
+  assert.equal(sock1._setNoDelay, true);
+  sock1.connect(common.PORT, function() {
+    var sock2 = new Socket({});
+    assert.equal(sock2._setNoDelay, null);
+    sock2.connect(common.PORT, function() {
+      assert.ok(setTrue);
```

Comment:
Also, wouldn't it already be true because of sock1?  Maybe `setTrue` should be a counter instead?